### DISCUSSION TITLE
nav: `Allowed memory size exhausted` with max_depth and recursion

### DISCUSSION THREAD

**Discussion by mstratman** [statamic/cms#7198] [Troubleshooting] [answered]
```html
  <ul>
    {{ nav:collection:c max_depth="2" }}
      <li>
        <a href="{{ url }}">{{ title }}</a>
        {{ if show_subcategories }}
          <ul>
            {{ *recursive children* }}
          </ul>
        {{ /if }}
      </li>
    {{ /nav:collection:c }}
  </ul>
```

With a category having a as much as 3 levels of hierarchy, when I omit max_depth it shows all 3 as expected.
When I include max_depth="3" it shows all 3 as expected.
When I include max_depth="1" it shows the string literal "{{ *recursive children* }}" - not sure if that's expected but it makes sense.

When I include max_depth="2", which should actually limit the depth and prevent the 3rd level from showing, I get this error:

`Symfony\Component\ErrorHandler\Error\FatalError: Allowed memory size of 134217728 bytes exhausted (tried to allocate 192512 bytes) in file /Users/mstratman/Projects/diypedals.net/diypedals.net/vendor/statamic/cms/src/View/Antlers/Parser.php on line 959`

Any tips to troubleshoot?

'show_categories' is a toggle field, if that matters.

**Comment by jasonvarga** [comment, marked as answer]
Wrap the recursion in a condition.

```antlers
{{ if children }}
<ul>{{ *recursive children* }}</ul>
{{ /if }}
```

or

```antlers
{{ if children && show_subcategories }}
```

**Reply by mstratman** [reply]
Excellent, thanks that fixed it.

**Reply by mstratman** [reply]
By the way, in my opinion you probably want to consider this a bug.
